src/CascFiles.cpp

```diff
@@ -1545,29 +1545,34 @@ DWORD CheckArchiveFilesDirectories(TCascStorage * hs)
     assert(hs->szDataPath == NULL);
     assert(hs->szIndexPath == NULL);
 
+    LPTSTR szDataPath;
+    LPTSTR szConfigPath;
+    LPTSTR szIndexPath;
+
     // Try all known subdirectories for the game data sub dirs
     for(size_t i = 0; i < _countof(DataDirs); i++)
     {
-        if((hs->szDataPath = CheckForDirectory(hs->szRootPath, DataDirs[i])) != NULL)
-        {
-            break;
-        }
-    }
+        if((szDataPath = CheckForDirectory(hs->szRootPath, DataDirs[i])) == NULL)
+            continue;
+
+        // If we found the data path, we also need to initialize the index path
 
-    // If we found the data path, we also need to initialize the index path
-    if(hs->szDataPath != NULL)
-    {
         // Check the config folder
-        if((hs->szConfigPath = CheckForDirectory(hs->szDataPath, _T("config"))) == NULL)
-            return ERROR_FILE_NOT_FOUND;
+        if((szConfigPath = CheckForDirectory(szDataPath, _T("config"))) == NULL)
```

Comment:
There will be a memory leak if:
1. Line 1555 succeeds the first time
2. Line 1561 fails the first time
3. Line 1555 succeeds the second time

Reply:
oops, should be fixed now